### PR TITLE
[MSHARED-1297] Prefer JDK features over plexus-utils, plexus-io

### PR DESCRIPTION
**Pull request by slawekjaranowski**
Following this checklist to help us incorporate your 
contribution quickly and easily:

 - [x] Make sure there is a [JIRA issue](https://issues.apache.org/jira/browse/MSHARED) filed 
       for the change (usually before you start working on it).  Trivial changes like typos do not 
       require a JIRA issue.  Your pull request should address just this issue, without 
       pulling in other changes.
 - [x] Each commit in the pull request should have a meaningful subject line and body.
 - [x] Format the pull request title like `[MSHARED-XXX] - Fixes bug in ApproximateQuantiles`,
       where you replace `MSHARED-XXX` with the appropriate JIRA issue. Best practice
       is to use the JIRA issue title in the pull request title and in the first line of the 
       commit message.
 - [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
 - [x] Run `mvn clean verify` to make sure basic checks pass. A more thorough check will 
       be performed on your pull request automatically.
 - [x] You have run the integration tests successfully (`mvn -Prun-its clean verify`).

If your pull request is about ~20 lines of code you don't need to sign an
[Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf) if you are unsure
please ask on the developers list.

To make clear that you license your contribution under 
the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0)
you have to acknowledge this by using the following check-box.

 - [x] I hereby declare this contribution to be licenced under the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0)

 - [x] In any other case, please file an [Apache Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

### CODE REVIEW

File: src/main/java/org/apache/maven/archiver/MavenArchiver.java [hunk 1]
```diff
@@ -800,6 +800,21 @@ public static Optional<Instant> parseBuildOutputTimestamp(String outputTimestamp
...
     }
 
+    private static boolean isNumeric(String str) {
+
+        if (str.isEmpty()) {
+            return false;
+        }
+
+        for (char c : str.toCharArray()) {
+            if (!Character.isDigit(c)) {
```
Comment (elharo) [inline src/main/java/org/apache/maven/archiver/MavenArchiver.java:810]: This is likely not what you want. It includes lots of non-ASCII digits like Devanagari digits. Long.parseLong won't work on these.

There might also be an issue with negative numbers.

**Review by elharo — CHANGES_REQUESTED** [review]

File: src/main/java/org/apache/maven/archiver/MavenArchiver.java (hunk 1 above)
Comment (slawekjaranowski) [reply src/main/java/org/apache/maven/archiver/MavenArchiver.java:810]: It is simple copy method from plexu-utils ...
https://github.com/codehaus-plexus/plexus-utils/blob/master/src/main/java/org/codehaus/plexus/util/StringUtils.java#L1749

I think it is enough for our case - if user provide a special chars Long.parseLong simply throw exception if needed.

Comment (slawekjaranowski) [reply src/main/java/org/apache/maven/archiver/MavenArchiver.java:810]: The same implementation is used in common-lang and commons-lang3

Comment (elharo) [reply src/main/java/org/apache/maven/archiver/MavenArchiver.java:810]: I guess this is within spec for parseBuildOutputTimestamp, as long as it's private and not used anywhere else.

**Review by elharo — APPROVED** [review]

**Pull request merged**